fathom/src/surface/elaboration.rs

```diff
@@ -760,25 +760,24 @@ impl<'interner, 'arena, 'error> Context<'interner, 'arena, 'error> {
 
     /// Reports an error if there are duplicate fields found, returning a slice
     /// of the labels unique labels and an iterator over the unique fields.
-    fn report_duplicate_labels<'fields, 'a>(
+    fn report_duplicate_labels<'fields, F>(
         &mut self,
         range: ByteRange,
-        fields: &'fields [((ByteRange, StringId), Term<'a, ByteRange>)],
-    ) -> (
-        &'arena [StringId],
-        impl Iterator<Item = &'fields ((ByteRange, StringId), Term<'a, ByteRange>)>,
-    ) {
+        fields: &'fields [F],
+        get_label: fn(&F) -> (ByteRange, StringId),
```

Comment:
Could this be a trait on `F` instead of passing a function?

Although looking at the closures that are actually passed perhaps it's not worth the ceremony.